### ISSUE TITLE
Error Destroying `aws_cloudformation_stack.email_subscription` Resource

### ISSUE THREAD

**Issue by smslattery** [localstack/localstack#3495]
# Type of request: This is a ...
[x] bug report
[ ] feature request

# Detailed description
When creating a `aws_cloudformation_stack.email_subscription` resource by running `terraform apply`, any tags associated with the resource are not being applied.

## Expected behavior
Created resource will be destroyed successfully.

## Actual behavior
```
Error: error waiting for CloudFormation Stack deletion: ValidationError: Stack with id arn:aws:cloudformation:us-east-1:000000000000:stack/some-name/aa33da8d does not exist
	status code: 400, request id: 6f9e0195
```

# Steps to reproduce
## Command used to start LocalStack
`IMAGE_NAME=localstack/localstack-full:latest localstack start` (image id: 7c1b3604a931)

## Client code (AWS SDK code snippet, or sequence of "awslocal" commands)
This was executed using terraform v0.12.6

```
$ terraform apply -auto-approve
aws_sns_topic.topic: Creating...
aws_sns_topic.topic: Creation complete after 0s [id=arn:aws:sns:us-east-1:000000000000:test-topic]
data.template_file.cloudformation_sns_subscription[0]: Refreshing state...
aws_cloudformation_stack.email_subscription: Creating...
aws_cloudformation_stack.email_subscription: Still creating... [10s elapsed]
aws_cloudformation_stack.email_subscription: Creation complete after 10s [id=arn:aws:cloudformation:us-east-1:000000000000:stack/some-name/aa33da8d]

Apply complete! Resources: 2 added, 0 changed, 0 destroyed.

Outputs:

email_subscription = {
  "name" = "some-email-subscription"
}
sns_topic = {
  "application_failure_feedback_role_arn" = ""
  "application_success_feedback_role_arn" = ""
  "arn" = "arn:aws:sns:us-east-1:000000000000:test-topic"
  "delivery_policy" = ""
  "display_name" = ""
  "http_failure_feedback_role_arn" = ""
  "http_success_feedback_role_arn" = ""
  "id" = "arn:aws:sns:us-east-1:000000000000:test-topic"
  "kms_master_key_id" = ""
  "lambda_failure_feedback_role_arn" = ""
  "lambda_success_feedback_role_arn" = ""
  "name" = "test-topic"
  "policy" = "{\"Version\": \"2008-10-17\", \"Id\": \"__default_policy_ID\", \"Statement\": [{\"Effect\": \"Allow\", \"Sid\": \"__default_statement_ID\", \"Principal\": {\"AWS\": \"*\"}, \"Action\": [\"SNS:GetTopicAttributes\", \"SNS:SetTopicAttributes\", \"SNS:AddPermission\", \"SNS:RemovePermission\", \"SNS:DeleteTopic\", \"SNS:Subscribe\", \"SNS:ListSubscriptionsByTopic\", \"SNS:Publish\", \"SNS:Receive\"], \"Resource\": \"arn:aws:sns:us-east-1:000000000000:test-topic\", \"Condition\": {\"StringEquals\": {\"AWS:SourceOwner\": \"000000000000\"}}}]}"
  "sqs_failure_feedback_role_arn" = ""
  "sqs_success_feedback_role_arn" = ""
}
template = [
  {
    "id" = "700135c1c582af3002091131a6eebba56106cec98755a772cd026b4fd1f0e8e5"
    "rendered" = "{\n  \"AWSTemplateFormatVersion\": \"2010-09-09\",\n\n  \"Resources\" : {\n    \"EmailSubscription\": {\n      \"Type\" : \"AWS::SNS::Subscription\",\n      \"Properties\" : {\n        \"Endpoint\": \"test@test.com\",\n        \"Protocol\": \"email\",\n        \"TopicArn\": \"arn:aws:sns:us-east-1:000000000000:test-topic\"\n      }\n    }\n  },\n\n  \"Outputs\" : {\n    \"SubscriptionARN\" : {\n      \"Description\" : \"Email SNS Topic ARN\",\n      \"Value\": { \"Ref\": \"EmailSubscription\" }\n    }\n  }\n}"
    "template" = "{\n  \"AWSTemplateFormatVersion\": \"2010-09-09\",\n\n  \"Resources\" : {\n    \"EmailSubscription\": {\n      \"Type\" : \"AWS::SNS::Subscription\",\n      \"Properties\" : {\n        \"Endpoint\": \"${endpoint}\",\n        \"Protocol\": \"email\",\n        \"TopicArn\": \"${topic_arn}\"\n      }\n    }\n  },\n\n  \"Outputs\" : {\n    \"SubscriptionARN\" : {\n      \"Description\" : \"Email SNS Topic ARN\",\n      \"Value\": { \"Ref\": \"EmailSubscription\" }\n    }\n  }\n}"
    "vars" = {
      "endpoint" = "test@test.com"
      "topic_arn" = "arn:aws:sns:us-east-1:000000000000:test-topic"
    }
  },
]

$ terraform destroy -auto-approve
aws_sns_topic.topic: Refreshing state... [id=arn:aws:sns:us-east-1:000000000000:test-topic]
data.template_file.cloudformation_sns_subscription[0]: Refreshing state...
aws_cloudformation_stack.email_subscription: Refreshing state... [id=arn:aws:cloudformation:us-east-1:000000000000:stack/some-name/aa33da8d]
aws_cloudformation_stack.email_subscription: Destroying... [id=arn:aws:cloudformation:us-east-1:000000000000:stack/some-name/aa33da8d]
aws_cloudformation_stack.email_subscription: Still destroying... [id=arn:aws:cloudformation:us-east-1:000000000000:stack/some-name/aa33da8d, 10s elapsed]

Error: error waiting for CloudFormation Stack deletion: ValidationError: Stack with id arn:aws:cloudformation:us-east-1:000000000000:stack/some-name/aa33da8d does not exist
	status code: 400, request id: 6f9e0195
```

## Terraform Config Files
main.tf

```
provider "aws" {
  region  = "us-east-1"
  access_key                  = "mock_access_key"
  secret_key                  = "mock_secret_key"
  s3_force_path_style         = true
  skip_credentials_validation = true
  skip_metadata_api_check     = true
  skip_requesting_account_id  = true
  endpoints {
    apigateway     = "http://${var.aws_host}:4566"
    cloudformation = "http://${var.aws_host}:4566"
    cloudwatch     = "http://${var.aws_host}:4566"
    dynamodb       = "http://${var.aws_host}:4566"
    ec2            = "http://${var.aws_host}:4566"
    es             = "http://${var.aws_host}:4566"
    firehose       = "http://${var.aws_host}:4566"
    iam            = "http://${var.aws_host}:4566"
    kinesis        = "http://${var.aws_host}:4566"
    lambda         = "http://${var.aws_host}:4566"
    route53        = "http://${var.aws_host}:4566"
    redshift       = "http://${var.aws_host}:4566"
    s3             = "http://${var.aws_host}:4566"
    secretsmanager = "http://${var.aws_host}:4566"
    ses            = "http://${var.aws_host}:4566"
    sns            = "http://${var.aws_host}:4566"
    sqs            = "http://${var.aws_host}:4566"
    ssm            = "http://${var.aws_host}:4566"
    stepfunctions  = "http://${var.aws_host}:4566"
    sts            = "http://${var.aws_host}:4566"
  }
}

variable "aws_host" {
  type          = "string"
  default       = "localhost"
}

resource "aws_sns_topic" "topic" {
  name = "test-topic"
}

data "template_file" "cloudformation_sns_subscription" {
  template = "${file("${path.module}/email-sns-stack.json.tpl")}"
  count = 1
  vars = {
    endpoint  = "test@test.com"
    topic_arn = "${aws_sns_topic.topic.arn}"
  }
}

resource "aws_cloudformation_stack" "email_subscription" {
  name          = "some-name"
  template_body = "${data.template_file.cloudformation_sns_subscription[0].rendered}"
  tags = {
    name = "some-email-subscription"
  }
}

output "email_subscription" {
  value = "${aws_cloudformation_stack.email_subscription.tags}"
}
```

email-sns-stack.json.tpl
```
{
  "AWSTemplateFormatVersion": "2010-09-09",

  "Resources" : {
    "EmailSubscription": {
      "Type" : "AWS::SNS::Subscription",
      "Properties" : {
        "Endpoint": "${endpoint}",
        "Protocol": "email",
        "TopicArn": "${topic_arn}"
      }
    }
  },

  "Outputs" : {
    "SubscriptionARN" : {
      "Description" : "Email SNS Topic ARN",
      "Value": { "Ref": "EmailSubscription" }
    }
  }
}
```

**Comment by smslattery** [comment]
Possibly related to: https://github.com/localstack/localstack/issues/3476

**Comment by whummer** [comment]
Thanks for reporting @smslattery . This should be fixed in #3581 - can you please pull the `latest` Docker image and give it another try? Please report here if the problem persists. Thanks

**Comment by smslattery** [comment]
Confirmed with the latest!
